benchexec/tools/theta.py

```diff
@@ -34,9 +34,16 @@ def cmdline(self, executable, options, task, rlimits):
         # Theta supports data race and unreach call
         if task.property_file:
             options += ["--property", task.property_file]
+        if isinstance(task.options, dict) and task.options.get("language") == "C":
+            data_model = task.options.get("data_model")
+            if data_model:
+                options += ["--architecture", data_model]
+
         return [executable, task.single_input_file] + options
 
     def determine_result(self, run):
+        if run.was_timeout:
+            return result.RESULT_TIMEOUT
```

Comment:
Note that you do not strictly need this change, BenchExec will anyway change the result to `TIMEOUT` if the run is marked as such. You can add this check if you want `determine_result` to not run or return some confusing result like "crashed" or so, though.

Reply:
Because we have the "Parsing OK" status which is almost always given early (right after parsing the input, before the analysis), I don't want to get many "TIMEOUT (Parsing OK)" results, which I think this fixes. At least that was the main motivation behind this.

Reply:
Yes, this is the use case for `was_timeout`. But you might also want to consider `was_terminated` then? Helps you to prevent `OUT OF MEMORY (Parsing OK)`.

Simply returning `result.RESULT_ERROR` from inside `if run.was_terminated:` is intended to work nicely (BenchExec will override the `ERROR` with the appropriate reason without producing something useless like `TIMEOUT (ERROR)`, but it will still produce something like `TIMEOUT (SEGMENTATION FAULT)` (which the current solution does not, besides handling only timeouts)).

Reply:
I see, that's a good point. I've added some code to handle `was_terminated` as well.
Let me know if you think this flow could still be improved!

Reply:
`was_timeout` is a subset of the cases covered by `was_terminated`, I recommend to just remove the former.

Reply:
Oh, sorry, I did not know that. Fixed.